src/sinks/datadog/logs/service.rs

```diff
@@ -103,12 +104,26 @@ impl LogApiService {
         uri: Uri,
         headers: IndexMap<String, String>,
     ) -> crate::Result<Self> {
-        let headers = validate_headers(&headers)?;
+        let user_provided_headers = validate_headers(&headers)?;
+
+        // Note that these headers cannot be overriden by the user.
+        let default_headers = &[
+            (CONTENT_TYPE.to_string(), "application/json".to_string()),
+            (
+                "DD-EVP-ORIGIN".to_string(),
+                crate::get_app_name().to_lowercase().replace(' ', "_"),
```

Comment:
I know it's only used in one place at the moment, but I could definitely see this being used in other places in the future if we wanted to pull it up.